dm/dm/config/task_converters.go

```diff
@@ -388,123 +400,210 @@ func SubTaskConfigsToTaskConfig(stCfgs ...*SubTaskConfig) *TaskConfig {
 	return c
 }
 
+// SubTaskConfigsToOpenAPITaskList gets openapi task from sub task configs.
+// subTaskConfigMap: taskName -> sourceName -> SubTaskConfig.
+func SubTaskConfigsToOpenAPITaskList(subTaskConfigMap map[string]map[string]*SubTaskConfig) []*openapi.Task {
+	taskList := []*openapi.Task{}
+	for _, subTaskConfigM := range subTaskConfigMap {
+		subTaskConfigList := make([]*SubTaskConfig, 0, len(subTaskConfigM))
+		for sourceName := range subTaskConfigM {
+			subTaskConfigList = append(subTaskConfigList, subTaskConfigM[sourceName])
+		}
+		taskList = append(taskList, SubTaskConfigsToOpenAPITask(subTaskConfigList))
+	}
+	return taskList
+}
+
 // SubTaskConfigsToOpenAPITask gets openapi task from sub task configs.
-func SubTaskConfigsToOpenAPITask(subTaskConfigMap map[string]map[string]SubTaskConfig) []openapi.Task {
-	taskList := []openapi.Task{}
-	for taskName, sourceMap := range subTaskConfigMap {
-		var oneSubtaskConfig SubTaskConfig // need this to get target db config
-		taskSourceConfig := openapi.TaskSourceConfig{}
-		sourceConfList := []openapi.TaskSourceConf{}
-		// source name -> filter rule list
-		filterMap := make(map[string][]*bf.BinlogEventRule)
-		// source name -> route rule list
-		routeMap := make(map[string][]*router.TableRule)
-		for sourceName, cfg := range sourceMap {
-			oneSubtaskConfig = cfg
-			oneConf := openapi.TaskSourceConf{
-				SourceName: sourceName,
-			}
-			if meta := cfg.Meta; meta != nil {
-				oneConf.BinlogGtid = &meta.BinLogGTID
-				oneConf.BinlogName = &meta.BinLogName
-				pos := int(meta.BinLogPos)
-				oneConf.BinlogPos = &pos
+func SubTaskConfigsToOpenAPITask(subTaskConfigList []*SubTaskConfig) *openapi.Task {
+	oneSubtaskConfig := subTaskConfigList[0] // need this to get target db config
+	taskSourceConfig := openapi.TaskSourceConfig{}
+	sourceConfList := []openapi.TaskSourceConf{}
```

Comment:
`TaskSourceConfig` and `TaskSourceConf` looks so similar, the later one is storing meta actually.